src/clarity-angular/data/tree-view/providers/tree-selection.provider.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {TreeSelectionService} from "./tree-selection.service";
+
+export function clrTreeSelectionProvider(existing: TreeSelectionService) {
```

Comment:
Should be named `clrTreeSelectionProviderFactory` or `clrTreeSelectionServiceFactory`.